apps/callscreen/js/handled_call.js

```diff
@@ -455,10 +440,26 @@ HandledCall.prototype.connected = function hc_connected() {
 
 HandledCall.prototype.disconnected = function hc_disconnected() {
   if (this._leftGroup) {
-    CallScreen.showStatusMessage({
-      id: 'caller-left-call',
-      args: { caller: this._cachedInfo.toString() }
-    });
+    if (typeof(this._cachedInfo) === 'string') {
+      CallScreen.showStatusMessage(this._cachedInfo);
+    } else if (this._cachedInfo.id) {
+      var id;
+
+      switch (this._cachedInfo && this._cachedInfo.id) {
+        case 'withheld-number': id = 'withheld-number-left-call';  break;
+        case 'voiceMail':       id = 'voicemail-left-call';        break;
+      }
+
+      CallScreen.showStatusMessage({
+        id: id,
+        args: { caller: this._cachedInfo ? this._cachedInfo.toString() : '' }
```

Comment:
I'm not sure if I understand this line.

In my patch I used formatValue for that scenario when you need to resolve an entity and then pass it as an argument as a string.

https://github.com/mozilla-b2g/gaia/pull/31104/files#diff-0cc741478a6ca5d2f75dd44f7f7ab18bR468

How is the toString supposed to work here?